### PR TITLE
feat(rust): implement ockam credential crate

### PR DESCRIPTION
**Pull request by mikelodder7**

### CODE REVIEW

**Review by jared-s — APPROVED** [review]

File: implementations/rust/ockam/ockam_credentials/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+
+All notable changes to this crate will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## 0.1.0
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/CHANGELOG.md:8]:
```suggestion
## 0.0.0
```

File: implementations/rust/ockam/ockam_credentials/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+exclude = [
+    "tests/**"
+]
+homepage = "https://ockam.io"
+keywords = []
+license = "Apache-2.0"
+name = "ockam_credentials"
+readme = "READNE.md"
+repository = "https://github.com/ockam-network/ockam"
+version = "0.1.0"
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/Cargo.toml:16]:
```suggestion
version = "0.0.0"
```

File: implementations/rust/ockam/ockam_credentials/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/CHANGELOG.md:9]:
```suggestion
## Added
```

File: implementations/rust/ockam/ockam_credentials/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+documentation = "https://docs.rs/ockam_credentials"
+edition = "2018"
+exclude = [
+    "tests/**"
+]
+homepage = "https://ockam.io"
+keywords = []
+license = "Apache-2.0"
+name = "ockam_credentials"
+readme = "READNE.md"
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/Cargo.toml:14]:
```suggestion
readme = "README.md"
```

File: implementations/rust/ockam/ockam_credentials/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+Ockam Credentials is a library for describing, creating, and verifying credentials
+on Ockam networks.
+
+## Usage
+
+Add this to your `Cargo.toml`:
+
+```
+[dependencies]
+ockam_credentials = "0.1.0"
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/README.md:17]:
```suggestion
ockam_credentials = "0.0.0"
```

File: implementations/rust/ockam/ockam_credentials/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+## Crate Features
+
+The `ockam_credentials` crate has a Cargo feature named `"std"` that is enabled by
+default. In order to use this crate in a `no-std` context this feature can
+disabled as follows
+
+```
+[dependencies]
+ockam_credentials = { version = "0.1.0", default-features = false, features = ["no-std"]  }
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/README.md:28]:
```suggestion
ockam_credentials = { version = "0.0.0", default-features = false, features = ["no-std"]  }
```

File: implementations/rust/ockam/ockam_credentials/src/error.rs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+impl CredentialError {
+    /// Integer code associated with the error domain.
+    pub const DOMAIN_CODE: u32 = 33_000;
+    #[cfg(feature = "std")]
+    /// Descriptive name for the error domain
+    pub const DOMAIN_NAME: &'static str = "OCKAM_CREDENTIALS";
+}
+
+#[cfg(feature = "std")]
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/src/error.rs:18]: Change to implement `From`

File: implementations/rust/ockam/ockam_credentials/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,205 @@
+/*
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/src/lib.rs:1]: Remove license header

File: implementations/rust/ockam/ockam_credentials/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    pub label: ByteString,
+    /// A longer description of the meaning of the attribute
+    #[serde(
+        serialize_with = "write_byte_string",
+        deserialize_with = "read_byte_string"
+    )]
+    pub description: ByteString,
+    /// The method that converts the attribute value to a cryptographic field element
+    #[serde(rename = "type")]
+    pub xtype: AttributeType,
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_credentials/src/lib.rs:201]:
```suggestion
    pub attribute_type: AttributeType,
```

**Review by jared-s — CHANGES_REQUESTED** [review]
For consistency with other crates, let's move the crate name to the singular form `ockam_credential`

**Review by jared-s — APPROVED** [review]

**Pull request merged**